marklogic/tests/test_marklogic.py

```diff
@@ -0,0 +1,207 @@
+import pytest
+from packaging import version
+
+from datadog_checks.base.stubs.aggregator import AggregatorStub
+from datadog_checks.marklogic import MarklogicCheck
+
+from .common import INSTANCE, INSTANCE_FILTERS, MARKLOGIC_VERSION
+from .metrics import (
+    FOREST_STATUS_SUMMARY_METRICS,
+    GLOBAL_METRICS,
+    HOST_STATUS_METRICS,
+    REQUESTS_STATUS_METRICS,
+    RESOURCE_STORAGE_FOREST_METRICS,
+    SERVER_STATUS_METRICS,
+    STORAGE_FOREST_METRICS,
+    STORAGE_HOST_METRICS,
+    TRANSACTION_STATUS_METRICS,
+)
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures("dd_environment")
+def test_check(aggregator):
+    # type: (AggregatorStub) -> None
+    check = MarklogicCheck('marklogic', {}, [INSTANCE])
+
+    check.check(INSTANCE)
+
+    tags = ['foo:bar']
+
+    for metric in GLOBAL_METRICS:
+        aggregator.assert_metric(metric, tags=tags)
+
+    storage_tag_prefixes = ['storage_path', 'host_name', 'host_id']
+    for metric in STORAGE_HOST_METRICS:
+        for tag in tags:
+            aggregator.assert_metric_has_tag(metric, tag)
+        for prefix in storage_tag_prefixes:
+            aggregator.assert_metric_has_tag_prefix(metric, prefix)
+    for metric in STORAGE_FOREST_METRICS:
+        for tag in tags:
+            aggregator.assert_metric_has_tag(metric, tag)
+        for prefix in storage_tag_prefixes + ['forest_id', 'forest_name']:
+            aggregator.assert_metric_has_tag_prefix(metric, prefix)
+
+    aggregator.assert_all_metrics_covered()
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures("dd_environment")
+def test_check_with_filters(aggregator):
+    # type: (AggregatorStub) -> None
+    check = MarklogicCheck('marklogic', {}, [INSTANCE_FILTERS])
+
+    check.check(INSTANCE_FILTERS)
+
+    for metric in HOST_STATUS_METRICS + SERVER_STATUS_METRICS + TRANSACTION_STATUS_METRICS:
+        aggregator.assert_metric(metric, count=1)
+    for metric in FOREST_STATUS_SUMMARY_METRICS + REQUESTS_STATUS_METRICS:
+        aggregator.assert_metric(metric, at_least=1)
+    for metric in STORAGE_HOST_METRICS:
+        aggregator.assert_metric(metric, count=2)
+        aggregator.assert_metric_has_tag(metric, 'forest_name:Security', count=1)
+    for metric in STORAGE_FOREST_METRICS:
+        aggregator.assert_metric(metric, count=10)
+
+    # Resource filter only
+    for metric in RESOURCE_STORAGE_FOREST_METRICS:
+        aggregator.assert_metric(metric, tags=['forest_name:Security'], count=1)
+    for metric in [
+        'marklogic.requests.query-count',
+        'marklogic.requests.total-requests',
+        'marklogic.requests.update-count',
+    ]:
+        aggregator.assert_metric(metric, tags=['server_name:Admin', 'group_name:Default'], count=1)
+
+    aggregator.assert_all_metrics_covered()
+
+    # Service checks
+    aggregator.assert_service_check('marklogic.can_connect', MarklogicCheck.OK, count=1)
+    aggregator.assert_service_check('marklogic.database.health', MarklogicCheck.OK, count=10)
```

Comment:
I think we should also assert service check tags and not use `count=10`.

There is `test_submit_health_service_checks`, but seems not to be a integration or e2e test.